src/test/java/org/elasticsearch/index/mapper/multifield/MultiFieldTests.java

```diff
@@ -446,4 +448,45 @@ public void testMultiFieldsInConsistentOrder() throws Exception {
             assertThat(field, equalTo(multiFieldNames[i++]));
         }
     }
+    
+    @Test
+    @SuppressWarnings("unchecked")
+    // The fielddata settings need to be in consistent order, else unneccesary mapping sync's can be triggered
+    public void testMultiFieldsFieldDataSettingsInConsistentOrder() throws Exception {
+        final String MY_MULTI_FIELD = "multi_field";
+        
+        // Possible fielddata settings
+        Map<String, Object> possibleSettings = new TreeMap<String, Object>();
+        possibleSettings.put("filter.frequency.min", 1);
+        possibleSettings.put("filter.frequency.max", 2);
+        possibleSettings.put("filter.regex.pattern", ".*");
+        possibleSettings.put("format", "fst");
+        possibleSettings.put("loading", "eager");
+        
+        // Generate a mapping with the a random subset of possible fielddata settings
+        XContentBuilder builder = jsonBuilder().startObject().startObject("type").startObject("properties")
+            .startObject("my_field").field("type", "string").startObject("fields").startObject(MY_MULTI_FIELD)
+            .field("type", "string").startObject("fielddata");
+        String[] keys = possibleSettings.keySet().toArray(new String[]{});
+        Collections.shuffle(Arrays.asList(keys));
+        for(int i = randomIntBetween(0, 4); i >= 0; --i)
+            builder.field(keys[i], possibleSettings.get(keys[i]));
+        builder.endObject().endObject().endObject().endObject().endObject().endObject().endObject();
+        
+        DocumentMapper docMapper = createIndex("test").mapperService().documentMapperParser().parse(builder.string());
```

Comment:
I think this test can be made stronger by actually checking the bytes are consistent, which is what we want to use. can we check the following?

```
        final DocumentMapperParser parser = createIndex("test").mapperService().documentMapperParser();
        DocumentMapper docMapper = parser.parse(builder.string());
        DocumentMapper docMapper2 = parser.parse(docMapper.mappingSource().string());
        assertThat(docMapper.mappingSource(), equalTo(docMapper2.mappingSource()));
```

It will cover all of this and more. We can then skip the rest (and also rename the test).